common/src/objects/workload_state.rs

```diff
@@ -168,6 +169,32 @@ pub enum ExecutionStateEnum {
     Removed,
 }
 
+// [impl->swdd~common-workload-state-transitions~1]
+impl ExecutionState {
+    pub fn transition(self, incoming: ExecutionState) -> ExecutionState {
+        match self.state {
+            ExecutionStateEnum::Stopping(StoppingSubstate::Stopping)
+            | ExecutionStateEnum::Stopping(StoppingSubstate::WaitingToStop) => match incoming.state
+            {
+                ExecutionStateEnum::Running(RunningSubstate::Ok)
+                | ExecutionStateEnum::Succeeded(SucceededSubstate::Ok)
+                | ExecutionStateEnum::Failed(FailedSubstate::ExecFailed)
+                | ExecutionStateEnum::Failed(FailedSubstate::Lost)
+                | ExecutionStateEnum::Failed(FailedSubstate::Unknown) => {
+                    log::trace!(
+                        "Skipping transition from '{}' to '{}' state.",
+                        self,
+                        incoming
+                    );
+                    self
+                }
+                _ => incoming,
+            },
+            _ => incoming,
+        }
+    }
+}
```

Comment:
The nested match statements can be removed by the following: 
```suggestion
impl ExecutionState {
    pub fn transition(self, incoming: ExecutionState) -> ExecutionState {
        match (&self.state, &incoming.state) {
            (
                ExecutionStateEnum::Stopping(StoppingSubstate::Stopping)
                | ExecutionStateEnum::Stopping(StoppingSubstate::WaitingToStop),
                ExecutionStateEnum::Running(RunningSubstate::Ok)
                | ExecutionStateEnum::Succeeded(SucceededSubstate::Ok)
                | ExecutionStateEnum::Failed(FailedSubstate::ExecFailed)
                | ExecutionStateEnum::Failed(FailedSubstate::Lost)
                | ExecutionStateEnum::Failed(FailedSubstate::Unknown),
            ) => {
                log::trace!(
                    "Skipping transition from '{}' to '{}' state.",
                    self,
                    incoming
                );
                self
            }
            _ => incoming,
        }
    }
}
```